pkg/utils/secrets/certificates.go

```diff
@@ -355,3 +357,45 @@ func GenerateCertificateAuthorities(k8sClusterClient kubernetes.Interface, exist
 
 	return generatedSecrets, certificateAuthorities, nil
 }
+
+// SelfGenerateTLSServerCertificate generates a new CA certificate and signs a server certificate with it. It'll return
+// the certificate objects and store the bytes in a temporary directory whose paths will also be returned.
+func SelfGenerateTLSServerCertificate(name string, dnsNames []string) (*Certificate, string, string, error) {
+	certificateDir, err := ioutil.TempDir("", "self-generated-server-certificates")
+	if err != nil {
+		return nil, "", "", err
+	}
+	certPath := filepath.Join(certificateDir, "tls.pem")
+	keyPath := filepath.Join(certificateDir, "tls-key.pem")
+
+	caCertificateConfig := &CertificateSecretConfig{
+		Name:       name,
+		CommonName: name,
+		CertType:   CACert,
+	}
+	caCertificate, err := caCertificateConfig.GenerateCertificate()
+	if err != nil {
+		return nil, "", "", err
+	}
+
+	certificateConfig := &CertificateSecretConfig{
+		Name:       name,
+		CommonName: name,
+		DNSNames:   dnsNames,
+		CertType:   ClientCert,
```

Comment:
Shouldn't it be server certificate type?
```suggestion
		CertType:   ServerCert,
```